lib/ruby_lsp/tapioca/addon.rb

```diff
@@ -106,6 +110,42 @@ def workspace_did_change_watched_files(changes)
           constants: constants,
         )
       end
+
+      private
+
+      sig { void }
+      def handle_gemfile_changes
+        return unless File.exist?(".git")
+
+        gemfile_status = %x(git status --porcelain Gemfile.lock).strip
+        return if gemfile_status.empty?
+
+        process_gemfile_changes
+      end
+
+      sig { returns(T.nilable(Integer)) }
+      def process_gemfile_changes
+        current_lockfile = File.read("Gemfile.lock")
+        snapshot_lockfile = File.read(GEMFILE_LOCK_SNAPSHOT) if File.exist?(GEMFILE_LOCK_SNAPSHOT)
+
+        unless snapshot_lockfile
+          $stdout.puts("Creating initial Gemfile.lock snapshot at #{GEMFILE_LOCK_SNAPSHOT}")
+          FileUtils.mkdir_p(File.dirname(GEMFILE_LOCK_SNAPSHOT))
+          File.write(GEMFILE_LOCK_SNAPSHOT, current_lockfile)
+          return
+        end
+
+        return if current_lockfile == snapshot_lockfile
+
+        T.must(@rails_runner_client).delegate_notification(
+          server_addon_name: "Tapioca",
+          request_name: "gem",
+          snapshot_lockfile: snapshot_lockfile,
```

Comment:
Will sending large lockfiles be an issue? Maybe getting timings on core is enough to answer this.

Reply:
On that note, do we have to send both lockfiles? What if we do the processing done in `server_addon.rb` here and send removed and changed gem names to the server instead? It's a possibility but I don't see a strong reason to do it unless there are memory/speed issues. Maybe only worth looking after we ship the feature.